src/main/java/mate/academy/springbootintro/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.springbootintro.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootintro.model.Book;
+import mate.academy.springbootintro.repository.SpecificationBuilder;
+import mate.academy.springbootintro.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+
+    private final SpecificationProviderManager<Book> specificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.title() != null && searchParameters.title().length > 0) {
+            spec = spec.and(specificationProviderManager.getSpecificationProvider("title")

```

Comment:
how about creating an enum to store search parameters instead of hardcoding them? It could be used here and specification provider classes.